### PR TITLE
Edits, updates, and refinements

### PR DESCRIPTION
**Pull request by jchodera**
It's been a while since we've cleaned up our best practices docs, so I'm in the process of doing this now.

### CODE REVIEW

**Comment by jchodera** [comment]
I've taken a quick edits pass to clean up some text and update some links.

File: LICENSING_GUIDELINES.md [hunk 1]
```diff
@@ -1,17 +1,19 @@
 # Licensing Guidelines
 
 When starting a new software project, there are many questions, from goals and milestones to implementation details.
-A critical detail is often glossed over: the license terms. While it may seem like a nightmare fit only for a lawyer,
-there are several open source licenses that are easy to understand and widely used by the community. To describe them, we'll separate
-licenses into two categories:
+A critical detail is often glossed over: the license terms.
+Explicitly specifying a license that gives your users permission to use, modify, and distribute your work is an important part of the process of reproducible science;
```
Comment (jchodera) [inline LICENSING_GUIDELINES.md]: Is this sentence too wordy?

File: VERSION_CONTROL.md [hunk 2]
```diff
@@ -20,37 +20,33 @@ There are 4 steps to modifying code:
...
-In a collaborative project, before merging, it is important to discuss with other colleagues whether the changes are appropriate.
-On [Github](http://github.com), this is done by means of a [pull request (PR)](https://help.github.com/articles/about-pull-requests/).
-
-Below is a basic description of a workflow that uses [Github](http://github.com).
-
-https://guides.github.com/introduction/flow/index.html
+In a collaborative project, before merging the branch you are working on, it is important to discuss with other colleagues whether the changes are appropriate.
+On [GitHub](http://github.com), this is done by means of a [pull request (PR)](https://help.github.com/articles/about-pull-requests/).
+This workflow may take some getting used to; to get the hang of it, check out [this tutorial](https://guides.github.com/introduction/flow/index.html).
+The process of creating branches can also be confusing at first; check out this [clever visual game](https://learngitbranching.js.org/?locale=en_US) to quickly get the idea of `git` branching.
```
Comment (andrrizzi) [inline VERSION_CONTROL.md]: This is pretty amazing!

**Comment by jchodera** [comment]
Thanks so much, @andrrizzi !
